logprep/processor/generic_adder/mysql_connector.py

```diff
@@ -47,25 +53,51 @@ def __init__(self, sql_config: dict, logger: Logger):
         """
         self._logger = logger
 
-        self.connection = db.connect(
-            user=sql_config["user"],
-            password=sql_config["password"],
-            host=sql_config["host"],
-            database=sql_config["database"],
-            port=sql_config.get("port", 3306),
-        )
+        self.connection = None
+        self.cursor = None
 
-        self.cursor = self.connection.cursor()
+        self._sql_config = sql_config
 
         self.target_column = sql_config["target_column"]
         self._add_target_column = sql_config.get("add_target_column", False)
 
         self.table_name = sql_config["table"]
 
-        self._timer = sql_config.get("timer", 60 * 3)
+        self._db_check_interval = sql_config.get("timer", 60 * 3)
         self._last_check = 0
         self._last_table_checksum = None
 
+    def connect(self):
+        """Get connection to SQL database and cursor for database table"""
+        self.connection = db.connect(
+            user=self._sql_config["user"],
+            password=self._sql_config["password"],
+            host=self._sql_config["host"],
+            database=self._sql_config["database"],
+            port=self._sql_config.get("port", 3306),
+        )
+        self.cursor = self.connection.cursor()
+
+    def disconnect(self):
+        """Close connection to SQL database"""
+        self.connection.close()
+
+    def time_to_check_for_change(self) -> bool:
+        """Check if enough time has passed to check for a SQL table change.
+
+        Update the timer if it is time to check for a change.
+
+        Returns
+        -------
+        bool
+            True if a check should be performed, False otherwise.
+
+        """
+        check_change = time.time() - self._last_check >= self._db_check_interval
+        if check_change:
+            self._last_check = time.time()
+        return check_change
+
     def has_changed(self) -> bool:
```

Comment:
In my opinion, this function could be simplified. If the last checksum is None, it is also not equal to the current checksum and therefore should just return True. My suggestion would be to simplify in line with your doctoring by putting the False case first and in any other case return True and set the self._last_table_checksum = checksum.